omnigibson/utils/usd_utils.py

```diff
@@ -325,18 +469,46 @@ def in_contact(cls, prim_paths_a, prim_paths_b):
         """
         # Check if the contact tuple already exists in the cache; if so, return the value
         key = (tuple(prim_paths_a), tuple(prim_paths_b))
-        if key not in cls._CONTACT_CACHE:
+        if key not in self._CONTACT_CACHE:
             # In contact if any of the matrix values representing the interaction between the two groups is non-zero
-            cls._CONTACT_CACHE[key] = np.any(cls.get_impulses(prim_paths_a=prim_paths_a, prim_paths_b=prim_paths_b))
-        return cls._CONTACT_CACHE[key]
+            self._CONTACT_CACHE[key] = np.any(self.get_impulses(prim_paths_a=prim_paths_a, prim_paths_b=prim_paths_b))
+        return self._CONTACT_CACHE[key]
 
-    @classmethod
-    def clear(cls):
+    def clear(self):
         """
         Clears the internal contact matrix and cache
         """
-        cls._CONTACT_MATRIX = None
-        cls._CONTACT_CACHE = dict()
+        self._CONTACT_MATRIX = dict()
+        self._CONTACT_DATA = dict()
+        self._CONTACT_CACHE = dict()
+
+
+# Instantiate the RigidContactAPI
+RigidContactAPI = RigidContactAPIImpl()
```

Comment:
what's the reason behind making this instantiatable? Maybe to avoid boilerplate wrappers, we either make the implementation class protected `_RigidContactAPIImpl` or have all of these APIs be owned by the simulator, e.g.:, `og.sim.rigid_contact_api` to unify all references? idk what is the best solution here.

Reply:
@cgokmen

Reply:
Main reason here is to allow subclassing/instancing e.g. like the Gripper version below and not needing to worry about what's a class variable thats owned by the subclass and/or the parent. Let's open a todo item for simulator-based ownership and also add the underscore to the class name as proposed @hang-yin